internal/storage/elasticsearch/config/config.go

```diff
@@ -588,28 +628,79 @@ func GetHTTPRoundTripper(ctx context.Context, c *Configuration, logger *zap.Logg
 			return nil, err
 		}
 		httpTransport.TLSClientConfig = ctlsConfig
-		transport = httpTransport
 	}
 
-	token := ""
-	if c.Authentication.BearerTokenAuthentication.FilePath != "" {
-		if c.Authentication.BearerTokenAuthentication.AllowFromContext {
-			logger.Warn("Token file and token propagation are both enabled, token from file won't be used")
+	var (
+		tokenFn         func() string
+		scheme          string
+		overrideFromCtx bool
+	)
+
+	switch {
+	case c.Authentication.APIKeyAuthentication.FilePath != "" && c.Authentication.APIKeyAuthentication.AllowFromContext:
+		logger.Warn("Both API key file and context propagation are enabled - context token will take precedence over file-based token")
+	case c.Authentication.BearerTokenAuthentication.FilePath != "" && c.Authentication.BearerTokenAuthentication.AllowFromContext:
+		logger.Warn("Token file and token propagation are both enabled, token from file won't be used")
+	}
+
+	switch {
+	case c.Authentication.APIKeyAuthentication.FilePath != "":
+		var err error
+		tokenFn, err = auth.TokenProvider(
+			c.Authentication.APIKeyAuthentication.FilePath,
+			TokenReloadInterval,
+			logger,
+		)
+		if err != nil {
+			return nil, err
 		}
-		tokenFromFile, err := loadTokenFromFile(c.Authentication.BearerTokenAuthentication.FilePath)
+		scheme = "ApiKey"
+		overrideFromCtx = c.Authentication.APIKeyAuthentication.AllowFromContext
+
+	case c.Authentication.APIKeyAuthentication.AllowFromContext:
+		tokenFn = func() string { return "" }
+		scheme = "ApiKey"
+		overrideFromCtx = true
+
+	case c.Authentication.BearerTokenAuthentication.FilePath != "":
```

Comment:
it looks like there are a lot of possible permutations (4 different mechanisms of getting a token). Please document somewhere the priority order. Perhaps it should be a single log line when more than one mechanism is enabled that states the exact priority.